internal/pkg/cosign/tsa/signer.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsa
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/digitorus/timestamp"
+	"github.com/sigstore/cosign/internal/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/bundle"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/mutate"
+
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	tsaclient "github.com/sigstore/timestamp-authority/pkg/generated/client"
+	ts "github.com/sigstore/timestamp-authority/pkg/generated/client/timestamp"
+)
+
+func GetTimestampedSignature(sigBytes []byte, tsaClient *tsaclient.TimestampAuthority) ([]byte, error) {
+	requestBytes, err := createTimestampAuthorityRequest(sigBytes, crypto.SHA256, "")
+	if err != nil {
+		return nil, err
+	}
+	fmt.Println("Calling TSA authority ...")
+	params := ts.NewGetTimestampResponseParams()
+	params.SetTimeout(time.Second * 30)
```

Comment:
Can we reduce this, this seems like a long timeout?

Reply:
I believe this is the default timeout value used by the timestamp-cli.

Reply:
We should update that too. For the sigstore TSA we'll aim for <1s p99s, for third parties, <10 or <5 should be reasonable